docs/_chapters/230-manifest-annotations.md

```diff
@@ -52,6 +54,12 @@ The analysis of these files happens after the analyzer plugins have been run. Th
 
 Since the annotations & imports happen in the comments, it is not possible to diagnose any errors. If the comment does not match its regular expression, it will be silently ignored.
 
+
+**Note:** Make sure `biz.aQute.bnd.annotation` is on the classpath / buildpath which contains the `aQute.bnd.annotation.spi.ServiceProvider` annotation. Otherwise the annotation won't be processed and you might see a warning in the log like this: 
+
+`Unable to determine whether the meta annotation aQute.bnd.annotation.spi.ServiceProvider applied to type xyz provides bundle annotations as it is not on the project build path. If this annotation does provide bundle annotations then it must be present on the build path in order to be processed`
```

Comment:
Maybe this warning could be more explicit? I don't see it in my build, it didn't log it. I had to really debug the code to come to it (see screenshot, it doesnt show any output)

<img width="1368" alt="Screenshot 2024-10-05 at 10 32 12" src="https://github.com/user-attachments/assets/0f6f3e0c-0fd7-4a8b-acc6-7adec7e158c1">
tput)

Doing this, would also make it clear for the maven build, seeing as adding something to the build path is done by adding a `<dependency>`

Reply:
This is currently logged via Sl4fj at either warning or debug level 
https://github.com/bndtools/bnd/blob/master/biz.aQute.bndlib/src/aQute/bnd/osgi/AnnotationHeaders.java#L336

Do you see it when you enable debug logging in Maven? (not sure, but I think it is the `-X` option).

Or:
Could you enable `-pendantic:true` in the bnd section, to see if it gets logged as WARN?
![image](https://github.com/user-attachments/assets/b9473dc2-60aa-4be8-b95c-fcd1151fc101)
 
Not sure what to do about it. 

@pkriens any thoughts why this was logged as warning or debug? Should we log as error or info?

Reply:
Ah ok it seems git blame shows a bit of why:
https://github.com/bndtools/bnd/commit/b1e83609913418b3cd8ade38ab8b260b9dbf302a

Reply:
Hmm, so does that mean we could set the log only for the bnd-maven-plugin instead of baseline? Or what would be a good alternative / fix so that we know there is something missing from the buildpath?

Reply:
> log only for the bnd-maven-plugin instead of baseline

Yeah something like that... For building to me it sounds like a missing annotation is important to log more visible. 
But I am too unfamiliar with both plugins and the thinking behind it and where to distinguish between those cases.

Maybe bnd-maven-plugin could set the `-pendantic:true` automatically? and then it gets logged as `.warn()`. 
But not sure this makes sense/

Have you tested with `-pendantic:true` to see if the maven build outputs the log message then? @royteeuwen

Reply:
I added experimental commit https://github.com/bndtools/bnd/pull/6309/commits/11172b9aa4dbf387340fe67b2e76f11dddea2018 where I set `pendantic:true` in `BndMavenPlugin` only.

Could you please retest by [building bnd and also bnd-maven-plugin](https://github.com/bndtools/bnd/pull/6309#issuecomment-2391967913) and see if you see the log.warn if you do not have the `biz.aQute.bnd.annotation` on the buildpath?

@pkriens @bjhargrave @timothyjward pinging you for feedback. If you have objects, I have no problem reverting this commit. Other suggestions welcome.

Reply:
Didn't seem to change anything. If I set the debugger, it states that the builder warnings and errors are null (in `AbstractBndMavenPlugin` but it does show that the builder.annotationHeaders.loggedMissing contains the ServiceProvider:

<img width="665" alt="Screenshot 2024-10-05 at 18 57 18" src="https://github.com/user-attachments/assets/7d72f76b-99c3-4064-9c78-325218da2035">

Reply:
Sorry. In the meantime I got `mvnDebug` working too and checked out [your branch](https://github.com/orbinson/opentelemetry-osgi-wrappers/tree/feature/switch-to-bnd) to play around.

I noticed I set the `pendantic` property incorrectly. 
Checkout my new commit https://github.com/bndtools/bnd/pull/6309/commits/ae10e72df508e78dac01619ce12ecf0c18933147 

After that the build output of `mvn package` looks like:

![image](https://github.com/user-attachments/assets/8b4b9b72-5d53-41e3-964e-346c79e75678)

and the debugger when into this branch:

![image](https://github.com/user-attachments/assets/a22ade48-e3f8-484e-bcab-23525182fc25)

Could you please try again? @royteeuwen

Reply:
Much better :D! Sorry that you had to learn / get `mvn` up and running 🥲 ...

Just going to tag @kwin, any reason for you that the `bnd-maven-plugin` should not have `pedantic:true` by default (meaning that there will be warnings logged). You made the bug for `bnd-baseline-maven-plugin` but it also affected the `bnd-maven-plugin`